app/controllers/products_controller.rb

```diff
@@ -45,18 +70,33 @@ def destroy
     end
   end
 
+  def get_category_children
+    @category_children = Category.find_by(name: "#{params[:parent_name]}", ancestry: nil).children
+  end
+
+  def get_category_grandchildren
+    @category_grandchildren = Category.find("#{params[:child_id]}").children
+  end
+
+
   private
+  
   def product_params
-    params.require(:product).permit(
-                            :name, :explain, :price, 
-                            :category_id, :brand_id, :condition_id, :deliveryday_id, :prefecture_id, :burden_id,
-                            images_attributes:[:image, :_destroy, :id]
-                             ).merge(
-                              user_id: current_user.id, exhibition_id: 2
-                              )
+    params.require(:product).permit(:name, :explain, :price, :brand_id, :condition_id, :deliveryday_id, :prefecture_id, :burden_id, :exhibiting, images_attributes:[:image, :_destroy, :id]).merge(user_id: current_user.id, exhibition_id: 2)
+  end
+
+  def product_params_update
+    params.require(:product).permit(:name, :explain, :price, :category_id, :brand_id, :condition_id, :deliveryday_id, :prefecture_id, :burden_id, :exhibiting, images_attributes:[:image, :_destroy, :id]).merge(user_id: current_user.id, exhibition_id: 2)
   end
 
+
   def set_product
     @product = Product.find(params[:id])
   end
+
+  def category_id_params
+    category = params.permit(:category_id)
+    @product[:category_id] = category[:category_id]
+  end
```

Comment:
@productに:category_idを加えているので``
```suggestion
def new
 @product[:category_id] =permit_category_id
end


private
  def permit_category_id
    category = params.permit(:category_id)[:category_id]
    return category
  end
```

とか動くならどうでしょうか
単なる提案です